vnext/Microsoft.ReactNative.Cxx/BoxedValue.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#ifndef MICROSOFT_REACTNATIVE_BOXEDVALUE
+#define MICROSOFT_REACTNATIVE_BOXEDVALUE
+
+#include <winrt/Microsoft.ReactNative.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+template <class T>
+struct BoxedValue : implements<BoxedValue<T>, IBoxedValue> {
+  template <class... TArgs>
+  BoxedValue(TArgs &&... args) noexcept : m_value(std::forward<TArgs>(args)...) {}
+
+  int64_t GetPtr() noexcept {
```

Comment:
intptr_t instead?

Reply:
This method implements the GetPtr method defined in the IBoxedValue. The MIDL 3.0 cannot use intptr. So, use the Int64 type there. If I use the intptr_t in the implementation, then it will cause runtime issues because the ABI requires it to be 64 bit value and intptr_t is 32 on 32 bit platforms.

---
In reply to: [424137927](https://github.com/microsoft/react-native-windows/pull/4884#discussion_r424137927) [](ancestors = 424137927)